mysys/CMakeLists.txt

```diff
@@ -134,7 +134,10 @@ ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
   ENDIF()
 ENDIF()
 
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64|powerpc64" OR CMAKE_SYSTEM_NAME MATCHES AIX)
+cmake_host_system_information(RESULT procname QUERY PROCESSOR_NAME)
+string(FIND ${procname} "PPC970" proc_is_ppc970)
+IF((proc_is_ppc970 EQUAL -1) AND
```

Comment:
On my system, `string()` populates `proc_is_ppc970` with `0` because `procname` is `PPC970` at this point, implying the offset to the substring is at the start of the string (it is).

Reply:
Please double check this, I don't want to break the ppc-optimized case but I don't have a newer chip to test on.

Reply:
As far as I understand, the SIMD implementation of CRC-32 is based on https://github.com/antonblanchard/crc32-vpmsum. The POWER v2.07 ISA was introduced in 2013. The IBM PowerPC 970 should implement the v2.03 ISA.

I suppose that @grooverdan had assumed that nobody would be running MariaDB on anything older than POWER8, which should already be 10 years old.

What operating system do you run? https://wiki.debian.org/ppc64el says that the minimum requirement is POWER8 (or the POWER v2.07 ISA). This suggests that a compile-time check (like the one you attempted) should be sufficient for older hardware and there might be no need to add a runtime check.

Reply:
> As far as I understand, the SIMD implementation of CRC-32 is based on https://github.com/antonblanchard/crc32-vpmsum. The POWER v2.07 ISA was introduced in 2013. The IBM PowerPC 970 should implement the v2.03 ISA.
> 
> I suppose that @grooverdan had assumed that nobody would be running MariaDB on anything older than POWER8, which should already be 10 years old.

I certainly assumed that `arch_ppc_probe` would return false if the hardware wasn't capable and therefore wouldn't SIGILL, even if the optimized code appeared in the compiled code.

Some lesser constraints existed for OSes that only supported POWER8+.

To complicate it more the Debian builds are chroots rather than a cross compile https://buildd.debian.org/status/package.php?p=mariadb where compile time checks may detect the OS of the host. Luckily powerpc is 32bit and somehow works.

> What operating system do you run? https://wiki.debian.org/ppc64el says that the minimum requirement is POWER8 (or the POWER v2.07 ISA). This suggests that a compile-time check (like the one you attempted) should be sufficient for older hardware and there might be no need to add a runtime check.

Reply:
fabaac86a18cfa026ca53b4a62e9edbf0bfd32d4 introduced a runtime check for something else that is based on trapping `SIGILL`. That could be an option here. We could have a compile-time check to enable the run-time detection only on operating systems that support older POWER ISA.